CONTRIBUTING.md

```diff
@@ -0,0 +1,315 @@
+| `media`      | Media files for the documentation or readme               |
+| `scripts`    | Scripts for simple tasks that are not part of the package |
+| `src/clinlp` | The source code for `clinlp`                              |
+| `tests`      | The tests for `clinlp`                                    |
+
+## Coding standards
+
+With `clinlp` we aim for code that is production-ready, well tested, scalable, maintainable, etc. We have a few guidelines that we follow to ensure that the codebase maintains a high quality.
+
+### General principles
+
+Please keep the following principles in mind when writing code:
+
+- Avoid repetitions, but refactor instead
+- Keep it simple, but not simpler
+- Only implement what is needed
+- Keep functions small and focused
+- Use descriptive names for variables, functions, classes, etc.
+- Apply SOLID principles and design patterns where applicable
+
+We fully acknowledge that writing production ready code is a skill that takes time to develop. We are happy to help work together, so please don't hesitate to reach out to us. This is especially true for scientific researchers who are working on something cool, but are new to software development.
+
+### Formatting and linting
+
+We use `ruff` for both formatting and linting. It is configured in `pyproject.toml`.
+
+The `ruff` formatter is a drop-in replacement for `black`. You can run it using the following command, which will format the codebase:
+
+```bash
+ruff format
+```
+
+The `ruff` linter checks for any common errors, that should be resolved before committing changes to the codebase. You can run it using the following command:
+
+```bash
+ruff lint
+```
+
+If any issues are found, some can be automatically fixed using the following command:
+
+```bash
+ruff lint --fix
+```
+
+### Dependencies
+
+We use `poetry` for managing dependencies. Adding a dependency is as straightforward as:
+
+```bash
+poetry add <package>
+```
+
+To keep `clinlp` lightweight, we only include dependencies that are strictly necessary in the base package, and use optional dependencies for additional functionality (e.g. transformers, metrics). You can add an optional dependency using:
+
+```bash
+poetry add --extras <extra> <package>
+```
+
+If you are adding specific development dependencies, please add them to a group such as `dev` or `docs` using:
+
+```bash
+poetry add --group <group> <package>
+```
+
+### Tests
+
+We use the `pytest` framework for testing. New code should preferably be accompanied by tests. We aim for a test coverage of at least 85%.
+
+You can run the tests locally by running:
+
+```bash
+pytest .
+```
+
+We preferably use the following `pytest` best practices:
+
+- Use fixtures to share setup code
+- Use parametrize to run the same test with different inputs
+- Use marks to skip tests, or to run tests with specific marks
+
+Additionally, we keep separation between integration, regression and unit tests:
```

Comment:
Maakt niet heel veel uit, maar is het niet logischer om de volgorde unit -> integration -> regression aan te houden?